__tests__/components/TransactionOptinPayoutDetails.spec.ts

```diff
@@ -0,0 +1,520 @@
+        });
+        test('returns false when inner transactions is empty', () => {
+            runBasicHasTransfersTests([], false);
+        });
+    });
+
+    describe('completed', () => {
+        const runBasicDisplayLabelTests = async (mockIsCompleted, { description, amount }) => {
+            // Arrange:
+            const wrapper = getTransactionOptinPayoutDetailsWrapper({
+                currentAccount: WalletsModel1,
+                transaction: createMockAggregateTransaction(),
+            });
+
+            // @ts-ignore
+            jest.spyOn(wrapper.vm, 'hasTransfers', 'get').mockReturnValue(true);
+
+            // @ts-ignore
+            jest.spyOn(wrapper.vm, 'completed', 'get').mockReturnValue(mockIsCompleted);
+
+            await wrapper.vm.$nextTick();
+
+            // Act + Assert:
+            expect(wrapper.find('.content-text.text-description').text()).toBe(description);
+            expect(wrapper.find('table .table-header-text').text()).toBe(amount);
+        };
+
+        test('returns true when transaction is confirmed', () => {
+            // Arrange:
+            const wrapper = getTransactionOptinPayoutDetailsWrapper({
+                currentAccount: WalletsModel1,
+                transaction: createMockAggregateTransaction(),
+            });
+
+            // Act + Assert:
+            // @ts-ignore
+            expect(wrapper.vm.completed).toBe(true);
+        });
+
+        test('returns false when transaction is unconfirmed', async () => {
+            // Arrange:
+            const wrapper = getTransactionOptinPayoutDetailsWrapper({
+                currentAccount: WalletsModel1,
+                transaction: new AggregateTransaction(
+                    NetworkType.TEST_NET,
+                    TransactionType.AGGREGATE_BONDED,
+                    1,
+                    mockDeadline,
+                    UInt64.fromUint(100),
+                    [],
+                    [],
+                    mockSignature,
+                    PublicAccount.createFromPublicKey(currentSigner.publicKey, NetworkType.TEST_NET),
+                ),
+            });
+
+            // Act + Assert:
+            // @ts-ignore
+            expect(wrapper.vm.completed).toBe(false);
+        });
+
+        test('renders completed amount and description text when completed is true', async () => {
+            await runBasicDisplayLabelTests(true, {
+                amount: 'The amount you have received:',
+                description: 'Success! Your opt-in payment is complete.',
+            });
+        });
+
+        test('renders pending amount and description text when completed is false', async () => {
+            await runBasicDisplayLabelTests(false, {
+                amount: 'The amount you will receive:',
+                description: 'Your opt-in payment is being processed. You will receive the XYM coins, when this transaction is confirmed',
+            });
+        });
+    });
```

Comment:
do we still need this logic anymore? all payouts are complete

Reply:
just check out the optin report, we didn't have any pending transactions actually.
I will remove it. 👍🏼

Reply:
please remove corresponding code from source

Reply:
patch fix 👍🏼